sematic/future_context.py

```diff
@@ -90,13 +113,33 @@ def context() -> SematicContext:
 
     Raises
     ------
-    RuntimeError:
+    NotInSematicFuncError:
         If this function is called outside the execution of a Sematic function.
     """
     global _current_context
     if _current_context is None:
-        raise RuntimeError(
+        raise NotInSematicFuncError(
             "context() must be called from within the execution of a Sematic function, "
             "in the root process that function was invoked from."
         )
     return _current_context
+
+
+class NotInSematicFuncError(RuntimeError):
+    pass
+
+
+class AbstractExternalResource:
```

Comment:
yes, added this comment:
```
# Note: this helps resolve circular imports. Otherwise we'd have:
# future_context -> external_resource -> future_context
# this approach cuts the first dependency in the above
```